### PR TITLE
storage: Improve behavior of size slider text and unit inputs

### PR DESCRIPTION
**Pull request by mvollmer**
Now the text and unit inputs stay constant when the user changes the other, and the slider moves accordingly. Previously, the slider would remain in place and the text input would change when the user changed the unit input.

I believe this is more natural: When people interact with the text and unit input, they very like start inputting the number and then they adjust the unit to what that number should be interpreted in.  In the old behavior, changing the unit would change the text so that the resulting value would still be the same. (Changing the unit from GB to MB would multiply the text by 1000.)

This was however masked by a bug: the text input did not actually change, although the SizeSLider component would pretend that it had. For example, when the unit was "GB", and the user typed in "1" into the text input and then changed the unit to "MB", a dialog would still receive 1 GB as the value of the slider.

With this change, the dialog would receive "1 MB" in this situation.

### CODE REVIEW

**Review by KKoukiou — DISMISSED** [review]
This would be probably easy to test to prevent regressing :) Maybe TODO item inline in the code as followup if you dont want now?

File: pkg/storaged/dialog.jsx [hunk 1]
```diff
@@ -1007,9 +1007,13 @@ class SizeSliderElement extends React.Component {
             text_val = cockpit.format_number(val / unit);
         }
 
-        const change_unit = (_, u) => this.setState({
-            unit: Number(u),
-        });
+        const change_unit = (_, u) => {
+            if (val.unit)
+                onChange({ text: val.text, unit: Number(u) });
```
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:1012]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19929-20240202-124556-94150a36-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by mvollmer** [comment]
> This would be probably easy to test to prevent regressing :) 

Yes, the existing test got broken by this, so let's update it.

File: pkg/storaged/dialog.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:1012]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19929-20240205-144049-d233f021-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by mvollmer** [comment]
fedora-rawhide seems to be broken rn: 
```
        Error: 
         Problem: conflicting requests
          - nothing provides python3-perf needed by tuned-2.21.0-2.fc40.noarch from testing-farm-tag-repository
          - nothing provides python3-perf needed by tuned-2.21.0-2.fc40.noarch from rawhide
```

File: pkg/storaged/dialog.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:1012]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19929-20240207-132441-76a6d7e0-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**